File: Serene/Serene.Web/Modules/Membership/Account/SignUp/SignUpForm.cs

namespace Serene.Membership
{
    using Serenity.ComponentModel;
    using System;
    using System.ComponentModel;

    [FormScript("Membership.SignUp")]
    public class SignUpForm
    {
        [Required(true)]
        public String DisplayName { get; set; }
        [EmailEditor, Required(true)]
        public String Email { get; set; }
        [EmailEditor, Required(true)]
        public String ConfirmEmail { get; set; }
        [PasswordEditor, Required(true)]
        public String Password { get; set; }
        [PasswordEditor, Required(true)]
        public String ConfirmPassword { get; set; }
    }
}

File: Serene/Serene.Web/Modules/ServerTypes/Membership/SignUpForm.ts
import { StringEditor, EmailAddressEditor, PasswordEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface SignUpForm {
    DisplayName: StringEditor;
    Email: EmailAddressEditor;
    ConfirmEmail: EmailAddressEditor;
    Password: PasswordEditor;
    ConfirmPassword: PasswordEditor;
}

export class SignUpForm extends PrefixedContext {
    static formKey = 'Membership.SignUp';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SignUpForm.init)  {
            SignUpForm.init = true;

            var w0 = StringEditor;
            var w1 = EmailAddressEditor;
            var w2 = PasswordEditor;

            initFormType(SignUpForm, [
                'DisplayName', w0,
                'Email', w1,
                'ConfirmEmail', w1,
                'Password', w2,
                'ConfirmPassword', w2
            ]);
        }
    }
}
